tools/kb_generator.py

```diff
@@ -0,0 +1,314 @@
+"""Module responsible for interacting with the OpenAI API to generate KB entries."""
+import ast
+import dataclasses
+import enum
+import io
+import json
+import os
+import pathlib
+from typing import Any
+
+import click
+import openai
+import tenacity
+from openai import openai_object
+from openai.api_resources import chat_completion
+
+
+if "PROD" in os.environ:
+    if "OPENAI_API_KEY" not in os.environ:
+        raise ValueError("OPENAI_API_KEY is not defined")
+
+    openai.api_key = os.getenv("OPENAI_API_KEY")
+    MODEL_NAME = "gpt-3.5-turbo"
+
+
+@dataclasses.dataclass
+class RiskRating(enum.Enum):
+    info = "INFO"
+    hardening = "HARDENING"
+    low = "LOW"
+    medium = "MEDIUM"
+    high = "HIGH"
+
+
+@dataclasses.dataclass
+class Platform(enum.Enum):
+    ios = "IOS"
+    android = "ANDROID"
+    multi = "MULTIPLATFORM"
+    common = "COMMON"
+    web = "WEB"
+
+
+@dataclasses.dataclass
+class Vulnerability:
+    name: str
+    risk_rating: RiskRating
+    platform: Platform
+
+
+@dataclasses.dataclass
+class KBEntry:
+    description: str
+    recommendation: str
+    meta: dict[str, Any]
+    vulnerability: Vulnerability | None = None
+
+
+PLATFORM_TO_PATH = {
+    "IOS": "MOBILE_CLIENT/IOS/",
+    "ANDROID": "MOBILE_CLIENT/ANDROID/",
+    "COMMON": "MOBILE_CLIENT/COMMON/",
+    "MULTIPLATFORM": "MOBILE_CLIENT/MULTIPLATFORM_JS/",
+    "WEB": "WEB_SERVICE/WEB/",
+}
+
+
+def dump_kb(kbentry: KBEntry) -> None:
+    path_prefix = pathlib.Path(
+        PLATFORM_TO_PATH[kbentry.vulnerability.platform],
+        f"_{kbentry.vulnerability.risk_rating}",
+    )
```

Comment:
Missing docstring. As a matter of fact, we need to add the workflow enforcing linting and typing.